source/extensions/quic_listeners/quiche/platform/quic_logging_impl.h

```diff
@@ -157,13 +179,23 @@ inline spdlog::logger& GetLogger() {
   return Envoy::Logger::Registry::getLog(Envoy::Logger::Id::quic);
 }
 
-inline bool IsLogLevelEnabled(QuicLogLevel level) { return level >= GetLogger().level(); }
+// This allows us to use QUICHE_CHECK(condition) from constexpr functions.
+#define QUICHE_IS_LOG_LEVEL_ENABLED(severity) quic::IsLogLevelEnabled##severity()
+#define QUICHE_IS_LOG_LEVEL_ENABLED_IMPL(severity)                                                 \
+  inline bool IsLogLevelEnabled##severity() { return quic::severity >= GetLogger().level(); }
+QUICHE_IS_LOG_LEVEL_ENABLED_IMPL(TRACE)
+QUICHE_IS_LOG_LEVEL_ENABLED_IMPL(DEBUG)
+QUICHE_IS_LOG_LEVEL_ENABLED_IMPL(INFO)
+QUICHE_IS_LOG_LEVEL_ENABLED_IMPL(WARNING)
+QUICHE_IS_LOG_LEVEL_ENABLED_IMPL(ERROR)
+QUICHE_IS_LOG_LEVEL_ENABLED_IMPL(DFATAL)
+inline bool constexpr IsLogLevelEnabledFATAL() { return true; }
```

Comment:
It's not obvious why IsLogLevelEnabledFATAL() can't be defined using the macro above?

Reply:
Because that wouldn't be constexpr, there's a comment above to that effect - do you have a suggestion for how I could improve the comment?